[recusal-list] Add strategy optional param

0x0-github opened the pull request:
Fixes # .

Changes proposed in this pull request:
- Addition of erc721-excluding-addresses strategy
- 
-

ChaituVR commented on src/strategies/erc721-excluding-addresses/index.ts:
```diff
@@ -0,0 +1,34 @@
+  );
+  return Object.fromEntries(
+    response.map((value, i) => [
+      addresses[i],
```
addresses will have different index than response, as you are filtering the addresses above

ChaituVR commented on src/strategies/erc721-excluding-addresses/index.ts:
```diff
@@ -0,0 +1,34 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+
+export const author = '0x0';
```
```suggestion
export const author = '0x0-github';
```

ChaituVR reviewed (changes requested):
Hi @0x0-github maybe we can modify https://snapshot.org/#/strategy/recusal-list this strategy to pass  a new param `strategy`  like this https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/anti-whale/index.ts#L69 will be more reusable

0x0-github commented:
Oh thanks indeed this would be better for future usages of anyone, can I just update it like that then ?
No need to mention my name then ?

0x0-github commented:
Updated the strategy (added optional strategy param and associated parameters as well), putted it to v 0.2.0 and removed the symbol parameter / definition (Doesn't seem to be used at all). Let me know if there's any issue / especially on the json definition file since I'm not a pro yet, some field may have more suitable config

ChaituVR commented on src/strategies/recusal-list/README.md:
```diff
@@ -1,14 +1,21 @@
     "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7"
-  ]
+  ],
+  "strategy": {
+    "name": "erc721",
+    "params": {
+      "address": "0x3B8CeB26f4FabACbD02b22caeceeb26D67E4013A",
+      "symbol": "MLZ"
+  }
```
```suggestion
  "strategy": {
    "name": "erc721",
    "params": {
      "address": "0x3B8CeB26f4FabACbD02b22caeceeb26D67E4013A",
      "symbol": "MLZ"
    }
  }
```

ChaituVR commented on src/strategies/recusal-list/index.ts:
```diff
@@ -1,21 +1,42 @@
+    );
+
+    return Object.fromEntries(
+      Object.entries(result).map(([address, _]) => [
```
```suggestion
      Object.entries(result).map(([address]) => [
```

ChaituVR commented on src/strategies/recusal-list/index.ts:
```diff
@@ -1,21 +1,42 @@
-      recusalList.includes(address.toLowerCase()) ? 0 : 1
-    ])
-  );
+  if (options.strategy.name) {
```
```suggestion
  if (options.strategy?.name) {
```

ChaituVR commented on src/strategies/recusal-list/index.ts:
```diff
@@ -1,21 +1,42 @@
+      space,
+      network,
+      provider,
+      addresses,
```
```suggestion
      recusalList,
```

ChaituVR commented on src/strategies/recusal-list/index.ts:
```diff
@@ -1,21 +1,42 @@
+      space,
+      network,
+      provider,
+      recusalList,
```
```suggestion
      addresses.filter((address: any) => !recusalList.includes(address.toLowerCase())),
```

0x0-github commented:
Sorry I just saw and finished to read your reviews. Thanks 🙏.

ChaituVR commented:
Deleting schema.json for now until we fix https://github.com/snapshot-labs/snapshot/issues/262
